integration_test/ops_agent_test.go

```diff
@@ -2091,6 +2093,386 @@ func TestPrometheusMetricsWithJSONExporter(t *testing.T) {
 	})
 }
 
+// Test the Histogram and Summary metric types using static testing files
+// The files will contain metrics in the right format and
+// hosted by a simple Python HTTP server so that the agent can scrape the metrics
+// The test will send two sets of metric points, to verify the cumulative metrics
+// are correctly received and processed
+func TestPrometheusMetricsHistogramAndSummary(t *testing.T) {
+	t.Parallel()
+	gce.RunForEachPlatform(t, func(t *testing.T, platform string) {
+		t.Parallel()
+		// TODO: Enable this test for all distros once the prometheus receiver is GA.
+		// For some reason, the featuregate, when set in the default systemd environment
+		// file, is not being picked up on centOS distros. This is a temporary workaround.
+		if gce.IsWindows(platform) || gce.IsCentOS(platform) || gce.IsRHEL(platform) {
+			t.SkipNow()
+		}
+		ctx, logger, vm := agents.CommonSetup(t, platform)
+
+		// 1. Upload the step one metric points
+		workDir := path.Join(workDirForPlatform(vm.Platform), "sample_data")
+		stepOneMetrics := "testdata/sample_histogram_summary_1.txt"
+		f, err := os.Open(stepOneMetrics)
+		if err != nil {
+			t.Fatal(err)
+		}
+		err = gce.UploadContent(ctx, logger, vm, f, path.Join(workDir, "data.json"))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// 2. Setup the Python env and start the http server
+		packages := []string{"python3"}
+		err = agents.InstallPackages(ctx, logger.ToMainLog(), vm, packages)
+		if err != nil {
+			t.Fatalf("failed to install %v with err: %s", packages, err)
+		}
+
+		// Run the setup script to start the Python http server
+		setupScript, err := os.ReadFile("testdata/setup_python_server.sh")
+		if err != nil {
+			t.Fatalf("failed to open setup script: %s", err)
+		}
+
+		env := map[string]string{"WORKDIR": workDir}
+		maxAttampts := 5
+		for attempt := 0; attempt < maxAttampts; attempt++ {
+			setupOut, err := gce.RunScriptRemotely(ctx, logger, vm, string(setupScript), nil, env)
+			// Since the script will start the processes in the background, the
+			// script should finish without error
+			if err != nil {
+				t.Fatalf("failed to run the http server in VM with err: %v, stderr: %s", err, setupOut.Stderr)
+			}
+			// Wait until the http server is ready
+			time.Sleep(30 * time.Second)
+			liveCheckOut, liveCheckErr := gce.RunRemotely(ctx, logger.ToMainLog(), vm, "", `curl "http://localhost:8000/data.json"`)
+			// We will retry when the Python HTTP server is not started:
```

Comment:
The retry was needed because the Python http server with nohup would randomly fail. The Python http server also different across distros and Python versions and make this test unstable. I have changed to use a go http server (since we can already set up go env in the vm) with systemd (to make sure the server starts, and logs to `journalctl_output.txt` for better debugging). From my testing it looks this is much more reliable than the previous approach.